File: questApplication/src/main/java/ru/nekit/android/qls/quest/answer/GroupWeightComparisonQuestAnswerChecker.java
package ru.nekit.android.qls.quest.answer;

import android.support.annotation.NonNull;

import java.util.List;

import ru.nekit.android.qls.quest.answer.common.AbstractGroupWeightComparisonQuestAnswerChecker;
import ru.nekit.android.qls.quest.common.Quest;
import ru.nekit.android.qls.quest.types.NumberSummandQuest;

public class GroupWeightComparisonQuestAnswerChecker
        extends AbstractGroupWeightComparisonQuestAnswerChecker<Integer> {

    @Override
    protected List<Integer> getGroupList(@NonNull Quest quest) {
        return ((NumberSummandQuest) quest).getLeftNodeAsList();
    }

}

File: questDomain/src/main/java/ru/nekit/android/qls/domain/answerChecker/GroupWeightComparisonQuestAnswerChecker.kt
package ru.nekit.android.qls.domain.answerChecker

import ru.nekit.android.qls.domain.answerChecker.common.AbstractGroupWeightComparisonQuestAnswerChecker
import ru.nekit.android.qls.domain.model.quest.NumberSummandQuest

class GroupWeightComparisonQuestAnswerChecker : AbstractGroupWeightComparisonQuestAnswerChecker<NumberSummandQuest>() {

    override fun getGroupList(quest: NumberSummandQuest): List<Int> = quest.leftNode.toList()

}